pkg/webhook/webhook.go

```diff
@@ -175,3 +196,112 @@ func (c *Webhook) validateHandlerFunc(review *admission_v1beta1.AdmissionReview)
 		Allowed: true,
 	}
 }
+
+func (c *Webhook) validateRelease(request *admission_v1beta1.AdmissionRequest, release shipper.Release) error {
+	var err error
+	if request.Operation == kubeclient.Create {
+		err = c.shouldBlockRelease(release)
+	} else if request.Operation == kubeclient.Update {
+		overrideRB, ok := release.Annotations[shipper.RolloutBlocksOverrideAnnotation]
+		if ok {
+			err = c.validateOverrideRolloutBlockAnnotation(overrideRB, release.Namespace)
+		}
+	}
+	return err
+}
+
+func (c *Webhook) validateApplication(request *admission_v1beta1.AdmissionRequest, application shipper.Application) error {
+	var err error
+	if request.Operation == kubeclient.Create {
+		err = c.shouldBlockApplication(application)
+	} else if request.Operation == kubeclient.Update {
+		overrideRB, ok := application.Annotations[shipper.RolloutBlocksOverrideAnnotation]
+		if ok {
+			err = c.validateOverrideRolloutBlockAnnotation(overrideRB, application.Namespace)
+		}
+	}
+	return err
+}
+
+func (c *Webhook) shouldBlockApplication(app shipper.Application) error {
+	ns := app.Namespace
+	overrideRB, ok := app.Annotations[shipper.RolloutBlocksOverrideAnnotation]
+	if !ok {
+		overrideRB = ""
+	}
+
+	return c.shouldBlockRollout(ns, overrideRB)
+}
+
+func (c *Webhook) shouldBlockRelease(release shipper.Release) error {
+	ns := release.Namespace
+	overrideRB, ok := release.Annotations[shipper.RolloutBlocksOverrideAnnotation]
+	if !ok {
+		overrideRB = ""
+	}
+
+	return c.shouldBlockRollout(ns, overrideRB)
+}
+
+func (c *Webhook) shouldBlockRollout(namespace string, overrideRB string) error {
+	var (
+		err          error
+		nsRBs, gbRBs []*shipper.RolloutBlock
+	)
+
+	nsRBs, gbRBs = c.existingRolloutBlocks(namespace)
+
+	if err = c.validateOverrideRolloutBlockAnnotation(overrideRB, namespace); err != nil {
+		return err
+	}
+
+	overrideRolloutBlock, eventMessage, err := rolloutblockUtil.ShouldOverrideRolloutBlock(overrideRB, nsRBs, gbRBs)
+	if err != nil {
+		return err
+	}
+
+	if overrideRolloutBlock {
+		return nil
+	}
+
+	return shippererrors.NewRolloutBlockError(eventMessage)
+}
+
+func (c *Webhook) existingRolloutBlocks(namespace string) ([]*shipper.RolloutBlock, []*shipper.RolloutBlock) {
```

Comment:
Consider returning a new structure like RolloutBlocks with attributes like: Local and Global